src/components/application_manager/src/request_controller.cc

```diff
@@ -364,9 +369,27 @@ void RequestController::onTimer() {
   SDL_AUTO_TRACE();
   SDL_DEBUG(
       "ENTER Waiting fore response count: " << waiting_for_response_.Size());
-  RequestInfoPtr probably_expired =
-      waiting_for_response_.FrontWithNotNullTimeout();
-  while (probably_expired && probably_expired->isExpired()) {
+  while(!stop_flag_) {
```

Comment:
@BrandonHe  in this realisation timer will work just like a thread. It is confuting to use timer for thread like functionality. 
If you think that threadlike functionality will be better, please rewrite request controller, remove timer and use thread. But I don't think that it is a good Idea.